pkg/provider/ecs/ecs.go

```diff
@@ -230,41 +237,67 @@ func (p *Provider) listInstances(ctx context.Context, client *awsClient) ([]ecsI
 		for _, cArn := range clustersArn {
 			clusters = append(clusters, *cArn)
 		}
+		for _, sArn := range servicesArns {
+			services = append(services, *sArn)
+		}
 	} else {
 		clusters = p.Clusters
+		services = p.Services
 	}
 
 	var instances []ecsInstance
 
 	logger.Debugf("ECS Clusters: %s", clusters)
+	logger.Debugf("ECS Services: %s", services)
 	for _, c := range clusters {
-		input := &ecs.ListTasksInput{
-			Cluster:       &c,
-			DesiredStatus: aws.String(ecs.DesiredStatusRunning),
-		}
-
+		cluster := c
+		var input *ecs.ListTasksInput
 		tasks := make(map[string]*ecs.Task)
-		err := client.ecs.ListTasksPagesWithContext(ctx, input, func(page *ecs.ListTasksOutput, lastPage bool) bool {
-			if len(page.TaskArns) > 0 {
-				resp, err := client.ecs.DescribeTasksWithContext(ctx, &ecs.DescribeTasksInput{
-					Tasks:   page.TaskArns,
-					Cluster: &c,
-				})
+		if len(services) > 0 {
+			for _, s := range services {
+				service := s
+				input = &ecs.ListTasksInput{
+					Cluster:       &cluster,
+					DesiredStatus: aws.String(ecs.DesiredStatusRunning),
```

Comment:
I don't see the point on setting it here without checking the flag, especially since it seems to be then ignored by listTasks

Reply:
Good point, not really being used. I'll make a new commit without this. Thanks!